src/apps/inventory/views.py

```diff
@@ -45,8 +45,38 @@ def get_queryset(self):
 
 class ProductListView(ListView):
     model = Product
-    paginate_by = 10
     template_name = "product_list.html"
+    context_object_name = "products"
+
+    def get_queryset(self):
+        queryset = Product.objects.all()
+        if self.request.GET:
+            filter_form = ProductFilterForm(self.request.GET)
+            if filter_form.is_valid():
+                name_query = filter_form.cleaned_data.get("name")
+                if name_query:
+                    queryset = queryset.filter(name__icontains=name_query)
+                category = filter_form.cleaned_data.get("category")
+                if category:
+                    queryset = queryset.filter(category=category)
+                min_price = filter_form.cleaned_data.get("min_price")
+                max_price = filter_form.cleaned_data.get("max_price")
+                if min_price is not None and max_price is not None:
+                    queryset = queryset.filter(price__range=(min_price, max_price))
+                order_by = self.request.GET.get("order_by")
+                if order_by:
+                    allowed_ordering_fields = ["name", "-name", "price", "-price"]
+                    if order_by in allowed_ordering_fields:
+                        queryset = queryset.order_by(order_by)
+                    else:
+                        queryset = queryset.order_by("-price")
+
+        return queryset
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["filter_form"] = ProductFilterForm(self.request.GET)
```

Comment:
Get some info on how to redefine extra context in Model Views

Reply:
I am not sure that i got all of it, but i think i am using it the right way now)